identity-account/src/account/account.rs

```diff
@@ -303,6 +306,58 @@ where
     Ok(())
   }
 
+  /// Encrypts the given `data` with the specified `algorithm`
+  ///
+  /// Diffie-Helman key exchange will be performed in case an [`KeyType::X25519`] is given.
+  pub async fn encrypt_data(
```

Comment:
That's a good idea! I suppose the issue is that we cannot feature gate it in `WasmStorage`, so I think you mean we will just unconditionally include it in `WasmStorage`.

Alternatively or specifically for bindings, we can explicitly allow implementations to return a `NotSupported` error or something, but the compile-time check is obviously preferred. It would be easier if we had a dedicated `StorageError` abstraction, too.